model.py

```diff
@@ -155,9 +155,23 @@ class Library(Base):
     # The official name of the library.
     name = Column(Unicode, index=True)
 
+    aliases = relationship("LibraryAlias", backref='library')
     service_areas = relationship('ServiceArea', backref='library')
 
     @classmethod
+    def for_name(cls, _db, name):
+        """Find a library whose name or alias matches the given name."""
+
+        # We allow for minor misspellings in the official name,
+        # but not in aliases (which are likely to be acronyms)
+        name_close_enough = func.levenshtein(func.lower(Library.name),
+                                             func.lower(name)) < 2
+        qu = _db.query(Library).outerjoin(Library.aliases).filter(
+            or_(name_close_enough, LibraryAlias.name.ilike(name))
+        )
```

Comment:
This looks good but I don't think it's quite enough.

If we have NYPL's official name as "The New York Public Library", a search for "new york public library" or "new york public" won't find it because the distance is too high. "New York Public Library" can be an alias, but then a search for "new york public libary" still won't work if aliases need exact matches.

I think allowing misspellings in the alias if the length is more than 6-ish could work well.